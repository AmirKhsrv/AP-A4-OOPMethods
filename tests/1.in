best_reviewer
best_book
credit 118
best_reviewer
recommend_first_approach 46
best_reviewer
best_book
credit 97
best_reviewer
credit 88
show_author_info 554
recommend_second_approach 112
show_sorted_shelf 33 read HolidayHorror
credit 1
show_sorted_shelf 33 read Romance
credit 114
recommend_second_approach 43
show_author_info 1681
show_sorted_shelf 15 read Nonfiction
