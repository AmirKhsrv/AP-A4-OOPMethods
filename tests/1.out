id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
0.007248
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
id: 392
Title: From the Mixed-Up Files of Mrs. Basil E. Frankweiler
Genre: Fiction
Author: E.L. Konigsburg
Reviews:
id: 356 Rating: 3 Likes: 0 Date: August 24 2018
id: 396 Rating: 2 Likes: 558 Date: January 16 2011
id: 677 Rating: 5 Likes: 3 Date: August 13 2018
id: 914 Rating: 5 Likes: 0 Date: June 3 2019
id: 1654 Rating: 0 Likes: 0 Date: February 6 2019
id: 1665 Rating: 5 Likes: 0 Date: November 21 2017
id: 1866 Rating: 0 Likes: 0 Date: December 27 2018
id: 2173 Rating: 2 Likes: 0 Date: February 17 2012
id: 2285 Rating: 5 Likes: 0 Date: March 28 2013
id: 2901 Rating: 5 Likes: 0 Date: July 20 2017
id: 4481 Rating: 5 Likes: 2 Date: May 21 2016
id: 5827 Rating: 4 Likes: 0 Date: November 25 2019
id: 5975 Rating: 1 Likes: 0 Date: March 28 2018
id: 6899 Rating: 5 Likes: 1 Date: February 27 2018
id: 8914 Rating: 1 Likes: 0 Date: July 12 2013
id: 9248 Rating: 5 Likes: 0 Date: October 4 2018
id: 9999 Rating: 5 Likes: 0 Date: March 18 2015
id: 10317 Rating: 4 Likes: 0 Date: April 18 2014
id: 10521 Rating: 4 Likes: 0 Date: September 13 2019
id: 10822 Rating: 5 Likes: 0 Date: May 11 2016
id: 11689 Rating: 5 Likes: 1 Date: January 27 2017
id: 12274 Rating: 5 Likes: 0 Date: August 25 2015
id: 12379 Rating: 5 Likes: 0 Date: June 23 2015
id: 12431 Rating: 0 Likes: 1 Date: July 25 2019
id: 12626 Rating: 3 Likes: 0 Date: September 18 2019
id: 13444 Rating: 0 Likes: 0 Date: June 26 2013
id: 13813 Rating: 3 Likes: 0 Date: May 27 2018
id: 14083 Rating: 3 Likes: 0 Date: October 26 2014
id: 14142 Rating: 4 Likes: 0 Date: December 27 2014
id: 14199 Rating: 3 Likes: 4 Date: October 12 2015
id: 14427 Rating: 1 Likes: 2 Date: June 15 2018
id: 16431 Rating: 5 Likes: 0 Date: October 28 2014
id: 16509 Rating: 5 Likes: 0 Date: March 17 2016
id: 16740 Rating: 3 Likes: 0 Date: April 26 2016
id: 16829 Rating: 0 Likes: 0 Date: October 15 2017
id: 17340 Rating: 3 Likes: 0 Date: November 13 2017
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
0.008433
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
0.009241
id: 554
Name: Joan He
Year of Birth: 1967
Place of Birth: Germany
Member Since: January 4 2019
Genres: Young Adult
Books:
id: 14 Title: Descendant of the Crane
id: 303
Title: The Good Luck Charm
Genre: Romance
Author: Helena Hunting
Reviews:
id: 99 Rating: 3 Likes: 8 Date: November 16 2018
id: 142 Rating: 4 Likes: 0 Date: January 20 2019
id: 949 Rating: 1 Likes: 1 Date: May 14 2017
id: 1209 Rating: 1 Likes: 0 Date: August 22 2013
id: 1350 Rating: 5 Likes: 2 Date: September 25 2018
id: 1595 Rating: 1 Likes: 0 Date: July 1 2013
id: 1747 Rating: 1 Likes: 0 Date: November 13 2018
id: 1846 Rating: 4 Likes: 0 Date: July 9 2016
id: 3401 Rating: 3 Likes: 0 Date: January 1 2016
id: 3889 Rating: 5 Likes: 0 Date: October 17 2017
id: 4887 Rating: 5 Likes: 3 Date: May 4 2018
id: 5127 Rating: 5 Likes: 0 Date: November 9 2010
id: 5574 Rating: 3 Likes: 0 Date: October 6 2014
id: 6137 Rating: 3 Likes: 0 Date: June 28 2017
id: 6480 Rating: 1 Likes: 1 Date: May 1 2011
id: 7559 Rating: 0 Likes: 7 Date: August 19 2017
id: 7913 Rating: 3 Likes: 0 Date: July 10 2019
id: 8048 Rating: 2 Likes: 0 Date: December 23 2018
id: 8160 Rating: 1 Likes: 0 Date: February 26 2017
id: 8310 Rating: 2 Likes: 0 Date: November 29 2019
id: 8626 Rating: 5 Likes: 0 Date: June 16 2014
id: 8683 Rating: 4 Likes: 0 Date: June 20 2013
id: 8858 Rating: 0 Likes: 0 Date: December 16 2015
id: 8862 Rating: 4 Likes: 0 Date: August 26 2018
id: 8903 Rating: 5 Likes: 0 Date: July 11 2018
id: 9182 Rating: 2 Likes: 0 Date: October 12 2018
id: 10427 Rating: 2 Likes: 0 Date: July 28 2018
id: 10432 Rating: 3 Likes: 0 Date: January 9 2018
id: 10490 Rating: 4 Likes: 0 Date: October 20 2017
id: 11047 Rating: 2 Likes: 24 Date: October 15 2010
id: 12102 Rating: 2 Likes: 3 Date: April 22 2018
id: 12147 Rating: 4 Likes: 2 Date: December 12 2019
id: 12633 Rating: 3 Likes: 0 Date: March 21 2016
id: 13302 Rating: 4 Likes: 0 Date: April 18 2015
id: 14111 Rating: 3 Likes: 1 Date: September 14 2016
id: 14698 Rating: 2 Likes: 0 Date: October 22 2019
id: 16634 Rating: 2 Likes: 0 Date: October 13 2018
id: 362
Title: A Season To Remember
Genre: Holiday
Author: Elizabeth Ellen Carter
***
id: 13
Title: Allegiant
Genre: Young Adult
Author: Veronica Roth
***
id: 318
Title: Blowing on Dandelions
Genre: Christian Fiction
Author: Miralee Ferrell
***
id: 400
Title: Dragon Bound
Genre: Fantasy
Author: Thea Harrison
***
id: 124
Title: Eating People is Wrong
Genre: Fiction
Author: Malcolm Bradbury
***
id: 135
Title: Fried Green Tomatoes at the Whistle Stop Cafe
Genre: Fiction
Author: Fannie Flagg
***
id: 388
Title: If I Have a Wicked Stepmother Where's My Prince?
Genre: Young Adult
Author: Melissa Kantor
***
id: 176
Title: Keturah and Lord Death
Genre: Fantasy
Author: Martine Leavitt
***
id: 35
Title: No Strings Attached
Genre: Romance
Author: Nicolette Day
***
id: 155
Title: Olive Kitteridge
Genre: Fiction
Author: Elizabeth Strout
***
id: 67
Title: Secret Daughter
Genre: Fiction
Author: Shilpi Somaya Gowda
***
id: 270
Title: Siddhartha
Genre: Classics
Author: Hermann Hesse
***
id: 342
Title: Something from the Nightside
Genre: Fantasy
Author: Simon R. Green
***
id: 8
Title: Starlet's Web
Genre: Young Adult
Author: Carla J. Hanna
***
id: 37
Title: Teaching My Mother How to Give Birth
Genre: Poetry
Author: Warsan Shire
***
id: 55
Title: Twelve Sharp
Genre: Mystery
Author: Janet Evanovich
***
id: 209
Title: Until Tuesday: A Wounded Warrior and the Golden Retriever Who Saved Him
Genre: Nonfiction
Author: Luis Carlos Montalván
***
id: 63
Title: When We Wake
Genre: Young Adult
Author: Karen Healey
***
0.007049
id: 35
Title: No Strings Attached
Genre: Romance
Author: Nicolette Day
***
id: 362
Title: A Season To Remember
Genre: Holiday
Author: Elizabeth Ellen Carter
***
id: 13
Title: Allegiant
Genre: Young Adult
Author: Veronica Roth
***
id: 318
Title: Blowing on Dandelions
Genre: Christian Fiction
Author: Miralee Ferrell
***
id: 400
Title: Dragon Bound
Genre: Fantasy
Author: Thea Harrison
***
id: 124
Title: Eating People is Wrong
Genre: Fiction
Author: Malcolm Bradbury
***
id: 135
Title: Fried Green Tomatoes at the Whistle Stop Cafe
Genre: Fiction
Author: Fannie Flagg
***
id: 388
Title: If I Have a Wicked Stepmother Where's My Prince?
Genre: Young Adult
Author: Melissa Kantor
***
id: 176
Title: Keturah and Lord Death
Genre: Fantasy
Author: Martine Leavitt
***
id: 155
Title: Olive Kitteridge
Genre: Fiction
Author: Elizabeth Strout
***
id: 67
Title: Secret Daughter
Genre: Fiction
Author: Shilpi Somaya Gowda
***
id: 270
Title: Siddhartha
Genre: Classics
Author: Hermann Hesse
***
id: 342
Title: Something from the Nightside
Genre: Fantasy
Author: Simon R. Green
***
id: 8
Title: Starlet's Web
Genre: Young Adult
Author: Carla J. Hanna
***
id: 37
Title: Teaching My Mother How to Give Birth
Genre: Poetry
Author: Warsan Shire
***
id: 55
Title: Twelve Sharp
Genre: Mystery
Author: Janet Evanovich
***
id: 209
Title: Until Tuesday: A Wounded Warrior and the Golden Retriever Who Saved Him
Genre: Nonfiction
Author: Luis Carlos Montalván
***
id: 63
Title: When We Wake
Genre: Young Adult
Author: Karen Healey
***
0.013648
id: 121
Title: Escaping Reality
Genre: Romance
Author: Lisa Renee Jones
Reviews:
id: 227 Rating: 1 Likes: 98 Date: July 19 2019
id: 410 Rating: 3 Likes: 279 Date: February 6 2016
id: 725 Rating: 4 Likes: 0 Date: November 17 2018
id: 1059 Rating: 5 Likes: 0 Date: November 3 2018
id: 1149 Rating: 0 Likes: 0 Date: October 29 2018
id: 1313 Rating: 3 Likes: 1 Date: April 28 2013
id: 1353 Rating: 5 Likes: 3 Date: December 16 2012
id: 1780 Rating: 4 Likes: 4 Date: September 26 2018
id: 1965 Rating: 0 Likes: 0 Date: February 15 2018
id: 2017 Rating: 2 Likes: 1 Date: April 28 2013
id: 2689 Rating: 5 Likes: 0 Date: March 14 2014
id: 2821 Rating: 1 Likes: 0 Date: March 14 2015
id: 4296 Rating: 2 Likes: 1 Date: November 8 2017
id: 4484 Rating: 2 Likes: 0 Date: May 15 2014
id: 4843 Rating: 0 Likes: 0 Date: February 21 2009
id: 4986 Rating: 4 Likes: 0 Date: September 24 2019
id: 5543 Rating: 4 Likes: 0 Date: November 14 2019
id: 6077 Rating: 1 Likes: 0 Date: August 1 2019
id: 6662 Rating: 1 Likes: 2 Date: December 12 2018
id: 7053 Rating: 2 Likes: 14 Date: June 17 2019
id: 7962 Rating: 4 Likes: 0 Date: May 8 2018
id: 9045 Rating: 2 Likes: 0 Date: April 9 2011
id: 9376 Rating: 1 Likes: 0 Date: May 12 2017
id: 9533 Rating: 4 Likes: 0 Date: March 9 2014
id: 10330 Rating: 5 Likes: 0 Date: May 16 2014
id: 10365 Rating: 1 Likes: 0 Date: September 2 2018
id: 10713 Rating: 1 Likes: 16 Date: January 28 2018
id: 11162 Rating: 1 Likes: 0 Date: January 24 2019
id: 11256 Rating: 3 Likes: 0 Date: February 13 2018
id: 11373 Rating: 3 Likes: 0 Date: June 27 2018
id: 11510 Rating: 4 Likes: 0 Date: February 19 2017
id: 12060 Rating: 4 Likes: 0 Date: August 19 2018
id: 12117 Rating: 2 Likes: 5 Date: May 7 2015
id: 12161 Rating: 5 Likes: 0 Date: November 2 2008
id: 12307 Rating: 4 Likes: 0 Date: May 21 2018
id: 14054 Rating: 2 Likes: 0 Date: April 9 2012
id: 14268 Rating: 2 Likes: 1 Date: September 9 2010
id: 14992 Rating: 2 Likes: 0 Date: June 14 2017
id: 15924 Rating: 5 Likes: 1 Date: August 11 2015
id: 16137 Rating: 5 Likes: 0 Date: August 18 2019
id: 16365 Rating: 4 Likes: 0 Date: September 13 2017
id: 16878 Rating: 4 Likes: 0 Date: October 7 2019
id: 1681
Name: Michael Connelly
Year of Birth: 1992
Place of Birth: Kosovo
Member Since: November 3 2009
Genres: Crime
Books:
id: 268 Title: Angels Flight
id: 314
Title: An Inconvenient Truth: The Planetary Emergency of Global Warming and What We Can Do About It
Genre: Nonfiction
Author: Al Gore
***
id: 206
Title: A Year to Remember
Genre: Romance
Author: Shelly Bell
***
id: 371
Title: All The Wrong Reasons
Genre: Romance
Author: J.L. Paul
***
id: 6
Title: Defective
Genre: Fiction
Author: Susan Sofayov
***
id: 167
Title: Shaedes of Gray
Genre: Fantasy
Author: Amanda Bonilla
***
id: 217
Title: Space Viking
Genre: Science Fiction
Author: H. Beam Piper
***
id: 210
Title: The Corsair's Captive
Genre: Science Fiction
Author: Ruby Dixon
***
id: 38
Title: The Golden Compass
Genre: Fantasy
Author: Philip Pullman
***
id: 10
Title: The Pillow Book
Genre: Cultural
Author: Sei Shōnagon
***
id: 137
Title: Touching Smoke
Genre: Fantasy
Author: Airicka Phoenix
***
id: 253
Title: You Belong to Me
Genre: Romance
Author: Karen      Rose
***
